bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -111,64 +104,104 @@ public WemoHandler(Thing thing, UpnpIOService upnpIOService, WemoHttpCall wemoHt
+        if (configuration.get(UDN) != null) {
+            logger.debug("Initializing WemoHandler for UDN '{}'", configuration.get(UDN));
+            UpnpIOService localservice = service;
+            if (localservice != null) {
+                localservice.registerParticipant(this);
+                host = getHost();
+            }
+            pollingJob = scheduler.scheduleWithFixedDelay(this::poll, 0, DEFAULT_REFRESH_INTERVALL_SECONDS,
+                    TimeUnit.SECONDS);
             updateStatus(ThingStatus.ONLINE);
         } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/config-status.error.missing-udn");
             logger.debug("Cannot initalize WemoHandler. UDN not set.");
         }
     }
 
     @Override
     public void dispose() {
-        logger.debug("WeMoHandler disposed.");
+        logger.debug("WemoHandler disposed for thing {}", getThing().getUID());
 
-        ScheduledFuture<?> job = refreshJob;
-        if (job != null && !job.isCancelled()) {
+        ScheduledFuture<?> job = this.pollingJob;
+        if (job != null) {
             job.cancel(true);
         }
-        refreshJob = null;
+        this.pollingJob = null;
         removeSubscription();
+        UpnpIOService localservice = service;
+        if (localservice != null) {
+            localservice.unregisterParticipant(this);
+        }
+    }
+
+    private void poll() {
+        synchronized (jobLock) {
+            if (pollingJob == null) {
+                return;
+            }
+            try {
+                logger.debug("Polling job");
+
+                if (host.isEmpty()) {
+                    host = getHost();
+                }
+                // Check if the Wemo device is set in the UPnP service registry
+                // If not, set the thing state to ONLINE/CONFIG-PENDING and wait for the next poll
+                if (!isUpnpDeviceRegistered()) {
+                    logger.debug("UPnP device {} not yet registered", getUDN());
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "@text/config-status.pending.device-not-registered [\"" + getUDN() + "\"]");
+                    synchronized (upnpLock) {
+                        subscriptionState = new HashMap<>();
+                    }
+                    return;
+                }
+                updateStatus(ThingStatus.ONLINE);
+                updateWemoState();
+                addSubscription();
+            } catch (Exception e) {
+                logger.debug("Exception during poll: {}", e.getMessage(), e);
+            }
+        }
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        logger.trace("Command '{}' received for channel '{}'", command, channelUID);
-
+        if (host.isEmpty()) {
+            if (service != null) {
+                host = getHost();
```

Comment:
host can still be empty after this call, which is currently not handled correctly. If you refactor getHost() a bit, you will be able to simply:

```
String localhost = getHost();
if (localHost.isEmpty()) {
    ...
    return;
}
```

I'll add a comment for `getHost()`.

Reply:
Changed